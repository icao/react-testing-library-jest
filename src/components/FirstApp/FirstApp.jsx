import  PropTypes  from "prop-types";

function FirstApp({ title, subtitle }) {
  return (
    <div className="app">
      <header className="app-header">
        <h1 className="title">{title}</h1>
        <h2 className="subtitle">{subtitle}</h2>
      </header>
    </div>
  );
}

FirstApp.propTypes = {
  title: PropTypes.string,
  suntitle: PropTypes.string,
}


export default FirstApp;
