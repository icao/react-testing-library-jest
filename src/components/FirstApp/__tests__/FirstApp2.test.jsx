import { render, screen } from "@testing-library/react";
import FirstApp from "../FirstApp";

describe("Pruebas en <FirstApp />", () => {
  const title = "Hola soy un titulo principal";
  const subtitle = "Hola soy un subtitulo secundario";

  test("Debe haces match con el snapshot", () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe mostrar el titulo", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByText(title)).toBeTruthy(); // [1] screen
    // screen.debug();
  });

  test("Debe mostrar el titulo en un H1", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    ); // [2] getByRole()
  });

  test("Debe mostrar el subtitulo enviado por props", () => {
    render(<FirstApp title={title} subtitle={subtitle} />);
    expect(screen.getAllByText(subtitle).length).toBe(3);
  });
});

/**
 * ------------------------------
 * [1]- screen
 * ------------------------------
 * screen es un objeto preligado al container que a su vez es retornado por el método render()
 *
 * Es asi como podemos hacer nuestas pruebas solo accediendo a l objeto screen sin depender de la declaración de un container.
 *
 * screen.deebug(): Donde debug() es un método que nos permite mostrar en ese momento de quiebre lo que tiene el snapshot rendereado.
 *
 * ------------------------------
 * [2]- getByRole()
 * ------------------------------
 * getByRole es otro método que nos provee React Testing Library para poder acceder a un elemento detro del DOM(el container o screen del snapshot).
 *
 * Podemos acceder a cualquier elemento especificando como primer parametro el tipo de elemento:
 *
 * getByRole([string_element])
 *
 * [string_element]: Elemento a buscar como un 'heading', 'button', etc.
 *
 * En casos como los headings debemos pasar como segundo parámetro, el nivel en especifico. Ejemplo: getByRole('heading', {level: 1})
 *
 * Lo que nos retornará el elemento en forma de objeto.
 *
 * */
