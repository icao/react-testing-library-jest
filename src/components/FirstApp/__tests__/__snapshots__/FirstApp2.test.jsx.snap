// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe haces match con el snapshot 1`] = `
<div>
  <div
    class="app"
  >
    <header>
      <h1
        data-testid="test-title"
      >
         
        Hola soy un titulo principal
         
      </h1>
      <h2
        data-testid="test-subtitle"
      >
        soy un subtitulo
      </h2>
      <h3>
        soy un subtitulo
      </h3>
      <p>
        soy un subtitulo
      </p>
    </header>
  </div>
</div>
`;
